src/tye/ApplicationBuilderExtensions.cs

```diff
@@ -122,7 +123,17 @@ public static Application ToHostingApplication(this ApplicationBuilder applicati
                 {
                     if (project.TargetFrameworks.Length > 1)
                     {
-                        throw new InvalidOperationException($"Unable to run {project.Name}. Multi-targeted projects are not supported.");
+                        if (project.BuildProperties.TryGetValue("TargetFramework", out var targetFramework))
+                        {
+                            if (!project.TargetFrameworks.Contains(targetFramework))
+                            {
+                                throw new InvalidOperationException($"Unable to run {project.Name}. The specified TargetFramework is not one of the existing TargetFrameworks in the project.");
+                            }
+                        }
+                        else
+                        {
+                            throw new InvalidOperationException($"Unable to run {project.Name}. Multi-targeted projects are not supported.");
```

Comment:
Change this exception message to say "specify a target framework in your tye.yaml or with --framework"

Reply:
I'll repro the same error with `dotnet run` and copy paste the error msg
is that fine for you ?

Reply:
Yep.

Reply:
done